datafusion/physical-expr/src/aggregate/sum.rs

```diff
@@ -315,6 +354,84 @@ pub(crate) fn sum(lhs: &ScalarValue, rhs: &ScalarValue) -> Result<ScalarValue> {
     })
 }
 
+pub(crate) fn add_to_row(
+    dt: &DataType,
+    index: usize,
+    accessor: &mut RowAccessor,
+    s: &ScalarValue,
+) -> Result<()> {
+    match (dt, s) {
+        // float64 coerces everything to f64
```

Comment:
🤔 I almost wonder how valuable supporting all these types are -- like I wonder if we can use `u64` or `i64` accumulators for all integer types and `f64` for floats and reduce the code. I don't think this PR is making things any better or worse, but it just seems like these type `match` statements are so common and repetitive

Reply:
Agree. We should clean this up by probably by checking type coercions.